### PR TITLE
[ADD] stock_account_unfuck: return product supplier/customer to real value of transaction

### PR DESCRIPTION
**Pull request by ebirbe**
This code is a migration of code introduced in https://github.com/Vauxoo/typ/pull/228 applying the coding recommendations made in that PR.

### CODE REVIEW

File: stock_account_unfuck/models/stock.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+        for move in self:
+            if move.product_qty == 0:
+                return
+            for quant in move.reserved_quant_ids:
+                if quant.qty <= 0:
+                    return
+            average_valuation_price = 0.0
+            qty_available = move.product_id.product_tmpl_id.qty_available
+            average_valuation_price = sum(
+                [quant.qty * quant.cost for quant in move.reserved_quant_ids])
```
Comment (hbto) [inline stock_account_unfuck/models/stock.py:24]: This is right when returning merchandise to Suppliers.
```
+            average_valuation_price = sum(
+                [quant.qty * quant.cost for quant in move.reserved_quant_ids])
```

When returning merchandise from `customer` or `transit`, the value to be used is the 
average price when the transaction was performed, i.e., `price_unit ` on
stock.move record.
https://github.com/odoo/odoo/blob/8.0/addons/stock/stock.py#L1811

**Review by hbto — CHANGES_REQUESTED** [review]

File: stock_account_unfuck/models/stock.py (hunk 1 above)
Comment (hbto) [reply stock_account_unfuck/models/stock.py:24]: @JulioSerna Do you agree on my suggestions?

**Comment by hbto** [comment]
UnitTests are to be added to this module for following cases:
- [ ] [Purchase Returns - Case Remaining Qty equals zero](https://github.com/vauxoo-dev/addons-vauxoo/blob/9a89032f1943f5d6fce3e00d84fc6949d211b3bb/stock_account_unfuck/README.md#odoos-approach-on-purchase-returns---case-remaining-qty-equals-zero)
- [ ] [Purchase Returns - Case Remaining Qty greater than zero](https://github.com/vauxoo-dev/addons-vauxoo/blob/9a89032f1943f5d6fce3e00d84fc6949d211b3bb/stock_account_unfuck/README.md#new-approach-on-purchase-returns---case-remaining-qty-greater-than-zero)
- [ ] [Sales Returns](https://github.com/vauxoo-dev/addons-vauxoo/blob/9a89032f1943f5d6fce3e00d84fc6949d211b3bb/stock_account_unfuck/README.md#new-approach-on-sales-returns)

**Comment by hbto** [comment]
@JulioSerna the accounting fix is not yet scheduled for this branch/PR?

File: stock_account_unfuck/models/stock.py (hunk 1 above)
Comment (JulioSerna) [reply stock_account_unfuck/models/stock.py:24]: 👍

**Pull request closed without merging**